File: DTOs/Enums/SystemType.swift
public enum SystemType: Codable, Equatable {
    case ios(version: String)   
    case android(version: String)
    case chrome(version: String)
    case edge(version: String)
    case firefox(version: String)
    case opera(version: String)
    case safari(version: String)
}

File: Android/shared/src/main/kotlin/shared/dto/enums/SystemType.kt
package shared.dto.enums

import kotlinx.serialization.Serializable
import shared.dto.protocols.Responsable

sealed class SystemType : Responsable {
    @Serializable
    data class Ios(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Android(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Chrome(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Edge(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Firefox(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Opera(
        val version: String,
    ) : SystemType()

    @Serializable
    data class Safari(
        val version: String,
    ) : SystemType()
}